### PR TITLE
Adding checksum .rsp for managed libraries

### PR DESCRIPTION
**Pull request by joperezr**
cc: @weshaggard 

related: dotnet/corefx#8479

### CODE REVIEW

**Comment by joperezr** [comment]
The way this will be consumed in corefx, is by adding 
```xml
<PropertyGroup>
  <CompilerResponseFile>$(ToolsDir)checksum.rsp</CompilerResponseFile>
</PropertyGroup>
```
To the root dir.props. That way, checksum.rsp will be used always, and for the specific case of System.Runtime, we will merge the contents of coreAssembly.rsp and checksum.rsp

**Comment by weshaggard** [comment]
I would prefer we just have a property that we enable as opposed to specifying the rsp file.

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/ReferenceAssemblies.targets [hunk 1]
```diff
@@ -37,4 +35,25 @@
...
+  <Target Name="MergeCoreAssemblyResponseFile">
+    <ItemGroup>
+      <ResponseFilesToMerge Include="$(CompilerResponseFile);$(MSBuildThisFileDirectory)coreAssembly.rsp" />
+    </ItemGroup>
+    <ReadLinesFromFile File="%(ResponseFilesToMerge.Identity)">
+      <Output TaskParameter="Lines" ItemName="ResponseFileLines" />
+    </ReadLinesFromFile>
+    <WriteLinesToFile File="$(IntermediateOutputPath)/resp.rsp" Lines="@(ResponseFileLines)" Overwrite="true" />
+    <PropertyGroup>
+      <CompilerResponseFile>$(IntermediateOutputPath)/resp.rsp</CompilerResponseFile>
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/ReferenceAssemblies.targets]: Please verify this but I think the compiler allows you to specify multiple rsp files so you don't need to merge them manually.

Comment (joperezr) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/ReferenceAssemblies.targets]: You are right. Looks like this works when you use a semicolon as a separator, I believe that in my testing I used a colon by mistake which led me to think it didn't work.

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.props [hunk 2]
```diff
@@ -12,4 +12,8 @@
   -->
   <Import Project="$(MSBuildThisFileDirectory)ReferenceAssemblies.props" Condition="'$(ExcludeReferenceAssembliesImport)'!='true'" />
 
+  <PropertyGroup>
+    <CompilerResponseFile Condition="'$(ExcludeChecsumParameterFromCsc)'!='true'">$(MSBuildThisFileDirectory)checksum.rsp</CompilerResponseFile>
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.props]: I would change this to.
`'$(CheckSumSHA256)'!='false'`

Comment (joperezr) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.props]: Done.

**Comment by weshaggard** [comment]
LGTM

**Pull request merged**